### PR TITLE
#25 use pks for queryset inside workers

### PR DESCRIPTION
**Pull request by codekiln**
addresses #25 use pks for queryset inside workers

### CODE REVIEW

File: django_elastic_migrations/__init__.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
         'This should be the python path to the elasticsearch client '
         'to use for indexing.')
 
-logger.debug("using DJANGO_ELASTIC_MIGRATIONS_ES_CLIENT = {}".format(settings.DJANGO_ELASTIC_MIGRATIONS_ES_CLIENT))
+# logger.debug("using DJANGO_ELASTIC_MIGRATIONS_ES_CLIENT = {}".format(settings.DJANGO_ELASTIC_MIGRATIONS_ES_CLIENT))
```
Comment (esmail) [inline django_elastic_migrations/__init__.py]: Keeping this around for later?

File: django_elastic_migrations/models.py [hunk 2]
```diff
@@ -365,14 +364,32 @@ def add_to_parent_docs_affected(self, num_docs):
...
+                try:
+                    with transaction.atomic():
+                        parent = (
+                            IndexAction.objects.select_for_update()
+                                .get(id=self.parent.id)
+                        )
+                        parent.docs_affected += num_docs
+                        parent_docs_affected = parent.docs_affected
+                        parent.save()
+                        successful = True
```
Comment (esmail) [inline django_elastic_migrations/models.py]: Not a problem, but looks like you could `break`

File: django_elastic_migrations/__init__.py (hunk 1 above)
Comment (codekiln) [reply django_elastic_migrations/__init__.py]: removed in 20a222df

File: django_elastic_migrations/models.py (hunk 2 above)
Comment (codekiln) [reply django_elastic_migrations/models.py]: What would be the value add for using `break` here?

File: django_elastic_migrations/indexes.py [hunk 3]
```diff
@@ -575,10 +575,20 @@ def generate_batches(cls, qs=None, batch_size=BATCH_SIZE, total_items=None, upda
         if qs is None:
             qs = cls.get_queryset()
 
+        update_index_action.add_log("START getting all ids to update")
+        try:
+            qs_ids = list(qs.values_list(cls.PK_ATTRIBUTE, flat=True))
+        except TypeError as e:
+            if "values_list() got an unexpected keyword argument 'flat'" in e:
```
[the remark this answers is not in the document]
Comment (codekiln) [reply django_elastic_migrations/indexes.py]: Haha, nice close reading. This is just in there for mongoengine users :)

**Pull request merged**